scripts: west_commands: runners: jlink: support pylink >= 0.14

gmarull opened the pull request:
pylink 0.14.0 changed the class variable where JLink DLL library name
(libjlinkarm) is stored. This patch adds support for new pylink
libraries while keeping backwards compatibility.

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/48626

stephanosio commented on scripts/west_commands/runners/jlink.py:
```diff
@@ -150,19 +150,7 @@ def jlink_version(self):
-
-            lib = Library(dllpath=os.fspath(Path(self.commander).parent /
-                                            libname))
+            lib = Library()
```
actually, it looks like this would introduce a behavioural change -- it used to load the JLink DLL from the path specified by the `--commander` argument (or the default), but that is no longer the case after this change.

This may lead to `jlink_version` reporting an incorrect version (different from that of the "commander") on a system with multiple JLink installations (not so uncommon).

gmarull replied:
fair point. changed strategy

gmarull committed:
scripts: west_commands: runners: jlink: support pylink >= 0.14

pylink 0.14.0 changed the class variable where JLink DLL library name
(libjlinkarm) is stored. This patch adds support for new pylink
libraries while keeping backwards compatibility.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>
